consensus/src/processes/pruning_proof/mod.rs

```diff
@@ -383,19 +382,25 @@ impl PruningProofManager {
         if proof.len() != self.max_block_level as usize + 1 {
             return Err(PruningImportError::ProofNotEnoughLevels(self.max_block_level as usize + 1));
         }
+        if proof[0].is_empty() {
```

Comment:
This change doesn't seem to be related to caches. Was it a bug that this wasn't here before, or did the cache changes introduce the need to add this check?

Reply:
I'm pretty sure it is the first option, but even if it is checked before this call it's better to check it here as well